app/controllers/email_confirmations_controller.rb

```diff
@@ -101,7 +101,6 @@ def mfa_update_conditions_met?
   end
 
   def setup_mfa_authentication
-    return if @user.totp_disabled?
     @form_mfa_url = mfa_update_email_confirmations_url(token: @user.confirmation_token)
```

Comment:
Should we move this to the original calling block as it's only called once? Is there a reason why this should be extracted into it's own method.